src/components/covariant-menu/covariant-menu.tsx

```diff
@@ -1,20 +1,57 @@
+import { useGlobal } from 'reactn'
 import React from 'react'
 
+import { OptionContainer } from '../option'
 import { MenuContainer } from '../menu-container'
+import { useNudgeDisplayMode } from '../../hooks'
+
 import {
-  DisplayModeOption,
-  HarpKeyOption,
-  RootPitchOption,
-  PozitionOption,
-} from '../../Menus/Options'
+  useNudgeHarpStrataByHarpKey,
+  useNudgeHarpStrataByPozition,
+  useNudgeHarpStrataByRootPitch,
+} from './hooks'
 
 export const CovariantMenu = (): React.ReactElement => {
+  const [activeHarpStrata] = useGlobal('activeHarpStrata')
+
+  const { harpKeyId } = activeHarpStrata
+  const nudgeHarpStrataByHarpKey = useNudgeHarpStrataByHarpKey()
+  const harpKeyOptionProps = {
+    title: 'Harp Key',
+    optionId: harpKeyId,
+    nudgeFunction: nudgeHarpStrataByHarpKey,
+  }
+
+  const { pozitionId } = activeHarpStrata
+  const nudgeHarpStrataByPozition = useNudgeHarpStrataByPozition()
+  const pozitionOptionProps = {
+    title: 'Position',
+    optionId: pozitionId,
+    nudgeFunction: nudgeHarpStrataByPozition,
+  }
+
+  const { rootPitchId } = activeHarpStrata
+  const nudgeHarpStrataByRootPitch = useNudgeHarpStrataByRootPitch()
+  const rootPitchOptionProps = {
+    title: 'Position Key',
+    optionId: rootPitchId,
+    nudgeFunction: nudgeHarpStrataByRootPitch,
+  }
+
+  const [activeDisplayMode] = useGlobal('activeDisplayMode')
+  const nudgeDisplayMode = useNudgeDisplayMode()
+  const displayModeOptionProps = {
+    title: 'Display',
+    optionId: activeDisplayMode,
+    nudgeFunction: nudgeDisplayMode,
+  }
```

Comment:
Arguably, each of these bundles of logic which result in a set of option properties could be bundled in to a hook. But that might be overkill if the code is sufficiently self explanatory in this form.

Reply:
This isn't necessary. The functionality is clear enough.